exporter/exporterbatcher/config.go

```diff
@@ -19,66 +20,70 @@ type Config struct {
-	// Deprecated. Ignored if SizeConfig is set.
+	// Deprecated: [v0.121.0] Ignored if SizeConfig is set.
 	MinSizeConfig `mapstructure:",squash"`
-	// Deprecated. Ignored if SizeConfig is set.
+	// Deprecated: [v0.121.0] Ignored if SizeConfig is set.
 	MaxSizeConfig `mapstructure:",squash"`
 }
 
+// SizeConfig sets the size limits for a batch.
 type SizeConfig struct {
 	Sizer SizerType `mapstructure:"sizer"`
 
-	MinSize int `mapstructure:"mix_size"`
+	// MinSize defines the configuration for the minimum size of a batch.
+	MinSize int `mapstructure:"min_size"`
+	// MaxSize defines the configuration for the maximum size of a batch.
 	MaxSize int `mapstructure:"max_size"`
 }
 
-// MinSizeConfig defines the configuration for the minimum number of items in a batch.
-// Experimental: This API is at the early stage of development and may change without backward compatibility
-// until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
+// Deprecated: [v0.121.0] use SizeConfig.
 type MinSizeConfig struct {
-	// MinSizeItems is the number of items (spans, data points or log records for OTLP) at which the batch should be
-	// sent regardless of the timeout. There is no guarantee that the batch size always greater than this value.
-	// This option requires the Request to implement RequestItemsCounter interface. Otherwise, it will be ignored.
-	MinSizeItems int `mapstructure:"min_size_items"`
+	// Deprecated: [v0.121.0] use SizeConfig.MinSize.
+	MinSizeItems *int `mapstructure:"min_size_items"`
 }
 
-// MaxSizeConfig defines the configuration for the maximum number of items in a batch.
-// Experimental: This API is at the early stage of development and may change without backward compatibility
-// until https://github.com/open-telemetry/opentelemetry-collector/issues/8122 is resolved.
+// Deprecated: [v0.121.0] use SizeConfig.
 type MaxSizeConfig struct {
-	// MaxSizeItems is the maximum number of the batch items, i.e. spans, data points or log records for OTLP.
-	// If the batch size exceeds this value, it will be broken up into smaller batches if possible.
-	// Setting this value to zero disables the maximum size limit.
-	MaxSizeItems int `mapstructure:"max_size_items"`
+	// Deprecated: [v0.121.0] use SizeConfig.MaxSize.
+	MaxSizeItems *int `mapstructure:"max_size_items"`
 }
 
-func (c Config) Validate() error {
-	if c.MinSizeItems < 0 {
-		return errors.New("min_size_items must be greater than or equal to zero")
+func (c *Config) Unmarshal(conf *confmap.Conf) error {
+	if err := conf.Unmarshal(c); err != nil {
+		return err
 	}
-	if c.MaxSizeItems < 0 {
-		return errors.New("max_size_items must be greater than or equal to zero")
+
+	if c.MinSizeItems != nil {
+		c.SizeConfig.MinSize = *c.MinSizeItems
 	}
-	if c.MaxSizeItems != 0 && c.MaxSizeItems < c.MinSizeItems {
-		return errors.New("max_size_items must be greater than or equal to min_size_items")
+
+	if c.MaxSizeItems != nil {
+		c.SizeConfig.MaxSize = *c.MaxSizeItems
```

Comment:
The doc comments above say:

```go
// Deprecated: [v0.121.0] Ignored if SizeConfig is set.
MinSizeConfig `mapstructure:",squash"`
// Deprecated: [v0.121.0] Ignored if SizeConfig is set.
MaxSizeConfig `mapstructure:",squash"
```

... whereas here we're overriding SizeConfig with Min/MaxSizeConfig if they're set. Behaviour/docs don't seem to line up.

Reply:
:+1: thanks!